plc4j/tools/connection-cache/src/main/java/org/apache/plc4x/java/utils/cache/ConnectionContainer.java

```diff
@@ -54,8 +62,19 @@ public synchronized Future<PlcConnection> lease() {
         }
         return connectionFuture;
     }
-
+    public synchronized void close(){
+        CompletableFuture<PlcConnection> leaseFuture;
+        while((leaseFuture = queue.poll())!=null){
+            leaseFuture.complete(null);
```

Comment:
Wouldn't it make more sense to complete exceptionally instead of returning null? We are currently expecting the result of a getConnection() to be a usable connection or an exception ... returning "null" sort of doesn't match that contract. I would change it to "completeExceptionally"

Reply:
I agree that this should throw exception